CMakeLists.txt

```diff
@@ -207,15 +216,16 @@ if (ONEDPL_BACKEND MATCHES "^(tbb|dpcpp|dpcpp_only)$")
             message(STATUS "Using a default device backend (*)")
         endif()
 
+        if (INTEL_LLVM_COMPILER AND INTEL_LLVM_COMPILER_VERSION VERSION_LESS 2023.1)
+            set(DEVICE_SELECTION_VARIABLE "SYCL_DEVICE_FILTER")
+            set(FPGA_DEVICE_TYPE "acc")
+        else()
+            set(DEVICE_SELECTION_VARIABLE "ONEAPI_DEVICE_SELECTOR")
+            set(FPGA_DEVICE_TYPE "fpga")
+        endif()
+
```

Comment:
Why this logic should be moved outside the `if`. It seems like it would define `DEVICE_SELECTION_VARIABLE` for all backends now even if `ONEDPL_DEVICE_TYPE` is not defined? Is it correct?

Reply:
Yes, exactly. I'll need `DEVICE_SELECTION_VARIABLE` later.